backend/pkg/rate/limit_http.go

```diff
@@ -188,17 +233,17 @@ func (m *HTTPLimiter) Reserve(r *http.Request) (Reservation, error) {
 			}
 			group := m.limiterGroups[b.String()]
 			if group != nil {
-				limiter = group
+				eventLimiter = group
 			}
 			b.Reset()
 		}
 	}
-	err := limiter.eventTemplate.Execute(&b, templateData)
+	err := eventLimiter.eventTemplate.Execute(&b, templateData)
 	if err != nil {
 		return nil, fmt.Errorf("error executing template for event ID: %w", err)
 	}
-	if limiter == nil {
-		return nil, nil
+	if eventLimiter == nil {
+		return okReservation{}, nil
 	}
```

Comment:
why do we first call `err := eventLimiter.eventTemplate.Execute(&b, templateData)` and then check `if eventLimiter == nil`?